src/udev/udevadm-control.c

```diff
@@ -136,11 +137,12 @@ static int adm_control(struct udev *udev, int argc, char *argv[]) {
                 case 't': {
                         int seconds;
 
-                        seconds = atoi(optarg);
-                        if (seconds >= 0)
-                                timeout = seconds;
-                        else
+                        rc = safe_atoi(optarg, &seconds);
```

Comment:
Then we could support the `infinity` option as `--timeout` value:

```c
seconds = s != NSEC_INFINITY ? (int) ((s + USEC_PER_SEC - 1) / USEC_PER_SEC) : INT_MAX;
```

I'll replace the `fprintf` calls + push it again!